Jenkinsfile

```diff
@@ -9,103 +9,115 @@ try {
     timeout(time: 1, unit: 'HOURS') {
       // Allocate a custom workspace to avoid having % in the path (it breaks ld)
       ws('/tmp/realm-java') {
-	stage('SCM') {
-	  checkout([
-		     $class: 'GitSCM',
-		    branches: scm.branches,
-		    gitTool: 'native git',
-		    extensions: scm.extensions + [
-		      [$class: 'CleanCheckout'],
-		      [$class: 'SubmoduleOption', recursiveSubmodules: true]
-		    ],
-		    userRemoteConfigs: scm.userRemoteConfigs
-		   ])
-	}
-
-	def buildEnv
-	def rosEnv
-	stage('Docker build') {
-	  // Docker image for build
-	  buildEnv = docker.build 'realm-java:snapshot'
-	  // Docker image for testing Realm Object Server
-	  def dependProperties = readProperties file: 'dependencies.list'
-	  def rosDeVersion = dependProperties["REALM_OBJECT_SERVER_DE_VERSION"]
-	  rosEnv = docker.build 'ros:snapshot', "--build-arg ROS_DE_VERSION=${rosDeVersion} tools/sync_test_server"
-	}
-
-	rosContainer = rosEnv.run('-v /tmp=/tmp/.ros')
-
-	try {
-          buildEnv.inside("-e HOME=/tmp " +
-			  "-e _JAVA_OPTIONS=-Duser.home=/tmp " +
-			  "--privileged " +
-			  "-v /dev/bus/usb:/dev/bus/usb " +
-			  "-v ${env.HOME}/gradle-cache:/tmp/.gradle " +
-			  "-v ${env.HOME}/.android:/tmp/.android " +
-			  "-v ${env.HOME}/ccache:/tmp/.ccache " +
-			  "--network container:${rosContainer.id}") {
-            stage('JVM tests') {
-              try {
-                withCredentials([[$class: 'FileBinding', credentialsId: 'c0cc8f9e-c3f1-4e22-b22f-6568392e26ae', variable: 'S3CFG']]) {
-                  sh "chmod +x gradlew && ./gradlew assemble check javadoc -Ps3cfg=${env.S3CFG}"
+        stage('SCM') {
+          checkout([
+                 $class: 'GitSCM',
+                branches: scm.branches,
+                gitTool: 'native git',
+                extensions: scm.extensions + [
+                  [$class: 'CleanCheckout'],
+                  [$class: 'SubmoduleOption', recursiveSubmodules: true]
+                ],
+                userRemoteConfigs: scm.userRemoteConfigs
+               ])
+        }
+
+        // Toggles for PR vs. Master builds.
+        // For PR's, we just build for arm-v7a and run unit tests for the ObjectServer variant
+        // A full build is done on `master`.
+        // TODO Once Android emulators are available on all nodes, we can switch to x86 builds
+        // on PR's for even more throughput.
+        def ABIs = null
```

Comment:
We must use empty string instead of `null`. `null` is not expanded to empty string in `GString`.